SnapWar/ViewControllers/SWAuthViewController.swift

```diff
@@ -1,10 +1,45 @@
 import UIKit
 
 class SWAuthViewController: UIViewController {
+    var usernameField = SWTextField()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        view?.backgroundColor = UIColor.green
+        view?.backgroundColor = SWColorType.white
+        setup()
+    }
+    
+    func setup() {
+        let formContainer = UIStackView()
+        view.addSubview(formContainer)
+        
+        formContainer.axis = .vertical
+        formContainer.spacing = 10
+        formContainer.distribution = .equalSpacing
+        formContainer.alignment = .fill
+        formContainer.backgroundColor = UIColor.blue
+            
+        let horizontalMargin:CGFloat = 10.0
+        formContainer.translatesAutoresizingMaskIntoConstraints = false
+        formContainer.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
+        formContainer.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor).isActive = true
+        formContainer.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor).isActive = true
+        formContainer.isLayoutMarginsRelativeArrangement = true
+        formContainer.layoutMargins = UIEdgeInsets(top: 0, left: horizontalMargin, bottom: 0, right: horizontalMargin)
+        
+        let keyboardTypes = [
+            UIKeyboardType.alphabet,
+            UIKeyboardType.numberPad,
+            UIKeyboardType.emailAddress
+        ]
+        for keyboardType in keyboardTypes {
+            let textField = SWTextField()
```

Comment:
not a feature requirement, but this might be a good opportunity for you to start using the delegate pattern. each text field can have a delegate, read up on https://developer.apple.com/documentation/uikit/uitextfielddelegate and figure out how to implement and read the content being inputted into each text field! if you want, you can get really fancy and hop from one input to the next with like return key or something